dapps/marketplace/src/components/DesktopUserActivation.js

```diff
@@ -84,89 +104,98 @@ class UserActivation extends Component {
       this.onStageChanged()
     }
 
-    this.updateStoredUserData({
-      firstName: this.state.firstName,
-      lastName: this.state.lastName,
-      avatar: this.state.avatar,
-      avatarUrl: this.state.avatarUrl
-    })
+    const hasUpdate =
+      this.state.firstName !== prevState.firstName ||
+      this.state.lastName !== prevState.lastName ||
+      this.state.avatar !== prevState.avatar ||
+      this.state.avatarUrl !== prevState.avatarUrl
+    if (hasUpdate) {
+      this.updateStoredUserData({
+        firstName: this.state.firstName,
+        lastName: this.state.lastName,
+        avatar: this.state.avatar,
+        avatarUrl: this.state.avatarUrl
+      })
+    }
   }
 
   render() {
     const {
       stage,
-      step,
       personalDataModal,
       shouldClosePersonalDataModal,
       txModal,
-      shouldCloseSignTxModal
+      shouldCloseSignTxModal,
+      confirmSkipModal,
+      shouldCloseConfirmSkipModal
     } = this.state
-    const { renderMobileVersion, hideHeader } = this.props
-
-    let stepHeader
-
-    const isMobile = this.props.ismobile === 'true'
-
-    const ModalComp = isMobile ? MobileModal : Modal
-
-    if (!hideHeader) {
-      stepHeader =
-        stage === 'ProfileCreated' ? null : (
-          <>
-            <h2 className="step-title">
-              {stage !== 'PublishDetail' && (
-                <fbt desc="UserActivation.addYourEmail">Add your email</fbt>
-              )}
-              {stage === 'PublishDetail' && (
-                <fbt desc="UserActivation.addNameAndPhoto">
-                  Add name and photo
-                </fbt>
-              )}
-            </h2>
-            <Steps steps={2} step={step} />
-          </>
-        )
-    }
+    const { renderMobileVersion } = this.props
 
     return (
       <div
         className={`user-activation${
           renderMobileVersion ? ' mobile' : ' desktop'
         }`}
       >
-        {stepHeader}
-        <div>{this[`render${stage}`]()}</div>
+        <div
+          className={`user-activation-content${
+            stage === 'RewardsSignUp' ? ' rewards-sign-up' : ''
+          }`}
+        >
+          {this[`render${stage}`]()}
+        </div>
         {personalDataModal && (
-          <ModalComp
+          <MobileModal
+            headerImageUrl="images/onboard/tout-header-image@3x.png"
             closeOnEsc={false}
             shouldClose={shouldClosePersonalDataModal}
             className="user-activation personal-data-modal"
             fullscreen={false}
             onClose={() =>
               this.setState({
-                personalDataModal: false,
-                shouldClosePersonalDataModal: false
+                personalDataModal: false
+                // shouldClosePersonalDataModal: false
```

Comment:
can we remove commented out code throughout?

Reply:
Will do that